pennylane/templates/subroutines/aqft.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains the template for AQFT.
+"""
+# pylint:disable=abstract-method,arguments-differ,protected-access
+
+import warnings
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import AnyWires, Operation
+
+
+class AQFT(Operation):
+    r"""AQFT(order, wires)
+    Apply an approximate quantum Fourier transform (AQFT).
+
+    The AQFT operation helps to reduce the number of ``ControlledPhaseShift`` operations required
+    for QFT by only using a maximum of ``order`` number of ``ControlledPhaseShift`` gates per qubit.
+
+    .. seealso:: :meth:`~.qft.QFT`.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 0
+    * Gradient recipe: None
+
+    Args:
+        order (int): the order of approximation
+        wires (int or Iterable[Number, str]]): the wire(s) the operation acts on
+
+    **Example**
+
+    The approximate quantum Fourier transform is applied by specifying the corresponding wires and
+    the order of approximation:
+
+    .. code-block::
+
+        wires = 3
+
+        dev = qml.device('default.qubit', wires=wires)
+
+
+        @qml.qnode(dev)
+        def circuit_aqft(basis_state):
+            qml.BasisState(basis_state, wires=range(wires))
+            qml.AQFT(order=1,wires=range(wires))
+            return qml.state()
+
+
+        circuit_aqft(np.array([1.0, 0.0, 0.0], requires_grad=False))
```

Comment:
This example may be confusing for those who are not familiar with `qml.BasisState`. I would suggest not to put parameters and put an X-gate for example